anesthetic/plot.py

```diff
@@ -974,6 +974,175 @@ def kde_plot_1d(ax, data, *args, **kwargs):
     return ans
 
 
+def nde_plot_1d(ax, data, *args, **kwargs):
+    """Plot a 1d marginalised distribution.
+
+    This functions as a wrapper around :meth:`matplotlib.axes.Axes.plot`, with
+    a normalising flow as a neural density estimation (NDE) provided by
+    `margarine.maf.MAF` or `margarine.clustered.clusterMAF`
```

Comment:
Getting the docs to work properly for an optional dependency seems like a pain, so I've copied the fastkde approach of just putting a link to margarine

Reply:
I don't understand, what was the issue here?

Reply:
There's probably a few layers to this, but part of it is that margarine only contains submodules (so `from margarine.maf import MAF` works but `import margarine; margarine.maf.MAF` does not, and I couldn't get to the bottom of how to refer to them properly.

You'll be able to see the error messages in the earlier checks

Reply:
Even changing the api hasn't made it straightforward - @lukashergt is the king when it comes to the docs

Reply:
To explain a bit more, the issue was that the docstring contained ``:class:`margarine.maf` ``, which tells autodoc to try and link to the documentation of a class called `maf` in a package called `margarine`, which autodoc wasn't able to find, though. Two issues here:

1. There is no dedicated documentation for the class  ``:class:`margarine.maf` ``, there is only documentation for the method [``:meth:`margarine.maf.MAF` ``](https://margarine.readthedocs.io/en/latest/functions.html#module-margarine.maf.MAF).
2. In order for autodoc to find the margarine documentation, you need to tell it where to look. In `docs/source/conf.py` you can find a dictionary that tells autodoc where to do that for matplotlib, pandas, and Co. You'll need to add the margarine readthedocs url to that dictionary in order to get the automatic linking to work.

Reply:
1. I think `:class:` is the right thing, since `margarine.maf` is a submodule, and `MAF` is a class inside that submodule.
2. I'd tried adding this and couldn't get it to work 😢

Reply:
Ah, it appears that the margarine docs don't build all the needed reference targets. I've created a PR htjb/margarine#54 that uses more of the autodoc power that should hopefully remedy that.

Reply:
Ahh okay I see! Thanks @lukashergt I will have a look at your PR on margarine asap.